### ISSUE TITLE
[Firefox 68] about:addons version numbers/compact no longer works.

### ISSUE THREAD

**Issue by MarkRH** [Aris-t2/CustomCSSforFx#214]
[Firefox 68] about:addons version numbers/compact no longer works.

1. Describe the issue
Using the following userContent.CSS:
```css
@import "./classic257/css/aboutaddons/addonlists_show_addon_version_number.css"; /**/
@import "./classic257/css/aboutaddons/addons_manager_alternative_appearance.css"; /**/
@import "./classic257/css/aboutaddons/addonlists_compact.css"; /**/
@import "./classic257/css/aboutconfig/aboutconfig_compact_appearance.css"; /**/
@import "./classic257/css/aboutaddons/description_page_alternative_content_order.css"; /**/
@import "./classic257/css/aboutaddons/recentupdates_category_always_visible.css"; /**/
@import "./classic257/css/aboutpreferences/preferences_alternative_appearance.css"; /**/
```
The version numbers do not show nor is the list compacted.

2. What should have happened instead?
Look similar to previous versions if possible.

3. Steps to reproduce the issue?
Run Firefox 68 with the above mentioned imports.

4. SCREENSHOTS with/without this projects files (drag & drop images into this post)
Before:
![image](https://user-images.githubusercontent.com/8035962/60927977-4d190400-a271-11e9-9f8e-5bb6058d1140.png)

After:
![image](https://user-images.githubusercontent.com/8035962/60927817-cfed8f00-a270-11e9-988c-5bd84f1ba383.png)

5. System information
- OS & OS version: Windows 7 Pro 64bit
- OS theme name: Classic
- DPI / HiDPI resolution: Normal/100%
- Firefox version: 68
- Firefox theme / lw-theme: Default
- This projects settings the issue occurs with: 
```css
@import "./classic257/css/aboutaddons/addonlists_show_addon_version_number.css"; /**/
@import "./classic257/css/aboutaddons/addons_manager_alternative_appearance.css"; /**/
@import "./classic257/css/aboutaddons/addonlists_compact.css"; /**/
```

**Comment by MarkRH** [comment]
Well, I managed to get the appearance back after seeing this thread: https://www.reddit.com/r/firefox/comments/cb0s0m/firefox_680_release_notes/etcc185/

Set `extensions.htmlaboutaddons.enabled` to `false`.

Now about:addons looks like:
![image](https://user-images.githubusercontent.com/8035962/60940433-0f33d400-a2a1-11e9-8c7d-46327edcaaef.png)

So, I guess that will do for now.. until that pref gets removed.

**Comment by Aris-t2** [comment]
This will happen sooner than later.
Restoring version number already does not work in current Fx70 Nightly.

**Comment by Pizzapops** [comment]
@MarkRH 
Sorry to see you are also afflicted with **MFS**  (Mozilla Frustration Syndrome). I am using Nightly 70 and your solution doesn't fully work. Yesterday, did a stop-gap single css file that does most of what I wanted ,
with the html config setting **True**. My version may be too compact for your taste. I am planning on using a grid or columns to eliminate vertical scroll. 
The three dot drop down menu is a big step backwards. Hovering over extension name does show ver#. 

![FxQ70_addons](https://user-images.githubusercontent.com/33039961/61067222-aea1b580-a3cc-11e9-9790-d0622bd3b588.jpg)
@Aris-t2  This was quick and dirty. Review my previous disclaimer!
 ```css
/* addonlists_compact70.css - Pizzpops and portions borrowed from Aris-t2 work */

:root {
    --section-width: 340px !important;
    --card-shadow: transparent !important;
    --card-shadow-hover: transparent !important;
}
body > div#main > div[current-view="list"] *,
body > div#main > div[current-view="list"] .addon.card,
body > div#main > div[current-view="list"] {
  padding: 0 0 1px 0!important;
  margin: 0 !important;
  min-height: 0 !important;
}

/* reduce category width*/
#categories {
  width: 100px !important;
  margin: 10px 0 0 !important;
}
/* adjust category icon position */
	richlistitem .category-icon {
	  -moz-margin-start: 13px !important;
	}
	
#category-discover > .category-icon {
    display:none !important;
}

.category-name,
.sidebar-footer-label{
 	display:none !important
}	
#search-label{
    display: none !important;
}
#header-search {
    margin-left: 20px !important;
    width: 340px !important;
}

#heading {
    margin-top: 4px !important;
    margin-bottom: 4px !important;
}
.heading-inner, 
#header-inner,
#header-name {
    width: 340px !important;
    }
    
.list-section-heading {
    color: #45A1FF !important;
}

addon-list[type="theme"] .card-heading-icon{
    width: 20px !important;
    height:  0px !important;
}
addon-list:not([type="theme"]) .card-heading-icon{
    width: 20px !important;
    height: 20px !important;
}

.addon.card:hover {
	background:#45A1FF !important;
	color:white !important;
    box-shadow: transparent!important;
}
 	
addon-list addon-card > .addon.card {
    width: 340px !important;
}
.addon-description {
    display: none !important;
}

.addon-card .card-heading-image {
    width: 340px !important;
    height: 46px !important;
}

**Comment by Pizzapops** [comment]
Can the three dot addon drop down menu be set to open on hover? One less mouse click!
[https://www.w3schools.com/howto/howto_css_dropdown.asp](url) I haven't played with this yet.
Simplifying the menu would be nice.  Disable, Remove and Options, maybe Report.
![hover_menu](https://user-images.githubusercontent.com/33039961/61160675-f6f0ce80-a4c5-11e9-9171-3df12b55643e.jpg)

**Comment by Aris-t2** [comment]
I think right now we only can manipulate CSS and no JS functions.

**Comment by Pizzapops** [comment]
I finally got columns to work.  Details needs to be fixed and this can break some web pages.
![2019-07-13_124135](https://user-images.githubusercontent.com/33039961/61174958-29093b80-a56d-11e9-9ff2-874113fd72da.jpg)
```css
/* addonlists_compact70.css - Pizzpops and portions borrowed from Aris-t2 work */

:root {
    --section-width: 360px !important;
    --card-shadow: transparent !important;
    --card-shadow-hover: transparent !important;
}
body > div#main > div[current-view="list"] *,
body > div#main > div[current-view="list"] .addon.card,
body > div#main > div[current-view="list"] {
  padding: 0 0 1px 0!important;
  margin: 0 !important;
  min-height: 0 !important;
}
body {
  columns: 360px 4 !important;
  column-rule: 4px double #45A1FF !important;
}
 
	/* reduce category width*/
#categories {
  width: 100px !important;
  margin: 10px 0 0 !important;
}
	/* adjust category icon position */
	richlistitem .category-icon {
	  -moz-margin-start: 13px !important;
	}
	
#category-discover > .category-icon {
    display:none !important;
}

.category-name,
.sidebar-footer-label{
 	display:none !important
}	
#search-label{
    display: none !important;
}
#header-search {
    margin-left: 20px !important;
    width: 360px !important;
}

#heading {
    margin-top: 4px !important;
    margin-bottom: 4px !important;
}
.heading-inner, 
#header-inner,
#header-name {
    width: 360px !important;
    }
    
.list-section-heading {
    color: #45A1FF !important;
}

addon-list[type="theme"] .card-heading-icon{
    width: 20px !important;
    height:  0px !important;
}
addon-list:not([type="theme"]) .card-heading-icon{
    width: 20px !important;
    height: 20px !important;
}

.addon.card:hover {
	background:#45A1FF !important;
	color:white !important;
    box-shadow: transparent!important;
}
 	
addon-list addon-card > .addon.card {
    width: 360px !important;
}
.addon-description {
    display: none !important;
}

.addon-card .card-heading-image {
    width: 360px !important;
    height: 46px !important;
}

**Comment by Aris-t2** [comment]
Looks great.

I ran some tests to at least rebuild "alternative appearance", some sort of "compact mode" and "more compact mode".

![001](https://user-images.githubusercontent.com/6013569/61175117-5922ff80-a5aa-11e9-99ef-fe99d27afa8c.png)

![002](https://user-images.githubusercontent.com/6013569/61175119-5b855980-a5aa-11e9-8eb9-1b52819d35cd.png)

![003](https://user-images.githubusercontent.com/6013569/61175120-5d4f1d00-a5aa-11e9-87fd-23411c408daf.png)

**Comment by Aris-t2** [comment]
Here is a dirty hack, in case you want to get the buttons out of the ... menu.

```css
.more-options-menu panel-list {
  display: block !important;
  visibility: visible !important;
  -moz-appearance: none !important;
  background: unset !important;
  border: unset !important;
  box-shadow: unset !important;
  -moz-margin-start: -540px !important;
  margin-top: -32px !important;
  min-width: unset !important;
  width: unset !important;
}

.more-options-menu,
.arrow {
  visibility: hidden !important;
}


.more-options-menu panel-list *:not([hidden]) {
  display: inline-flex !important;
  visibility: visible !important;
  -moz-box-orient: horizontal !important;
  -moz-appearance: button !important;
}
```

![1](https://user-images.githubusercontent.com/6013569/61175548-f5043980-a5b1-11e9-8213-540f85a9df19.png)

**Comment by Pizzapops** [comment]
@Aris-t2 
Things are going in the right direction. My about:addons columns settings break many web sites. How do I limit the body settings to about:addons? It puts columns in Github and many other web site and scrambles the pages.

**Comment by Aris-t2** [comment]
Normally there is no way to exclude the body tag from other pages, if it does not use unique indicators like ids or classes, but in this case you can attach your code to the internal sub page:

```css
@-moz-document  url-prefix(chrome://mozapps/content/extensions/aboutaddons.html) {
// code
}
```

**Comment by Aris-t2** [comment]
I think we are getting somewhere. Version number can be restored using CSS, but for active add-ons only.

![001](https://user-images.githubusercontent.com/6013569/61176936-d447de00-a5c9-11e9-8818-9cf253f8abda.png)

**Comment by Speravir** [comment]
I’d really love to see the version numbers.

Some additions for `about:addons`, though not for the numbers:

I have `addons_manager_alternative_appearance_fx67.css` activated. With this the addons bar was slightly too wide for my taste. I added a local user style
```css
.card.addon {
	margin-left: auto !important;
	margin-right: auto !important;
	width: 95vw !important;
}
```
I used first percent, but in Camp Firefox another user pointed to `vw`. I always forget about this. Note that `.card.addon` is a div.

Then for the disabled addons section I added this:

```css
/* deactivated addons & themes */
addon-list section[section="1"] {
	color: grey !important;
	filter: opacity(0.5) !important; 
}
```
Don’t ask me why I used `filter`. Actually, I started with thinking of grayscale for the addon images and left it this way.

And for the aero style this or a similar colour could be added, this changes the colour of the box-shadow on hovering:
```css
:root {
	--card-outline-color: rgb(154,189,255) !important; /* derivation of "cornflowerblue" */
}
```
